lib/Passwordless.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace WorkOS;
+
+/**
+ * Class Passwordless.
+ *
+ * This class facilitates the use of WorkOS Magic Link.
+ */
+class Passwordless
+{
+    /**
+     * Generates a passwordless session.
+     *
+     * @param string $email Email address of the user that the session is to be created for
+     * @param null|string URI to direct the user to user to upon authenticating through the passwordless link
+     * @param null|array $state Associative array containing state that will be returned from WorkOS as a json encoded string
+     * @param \WorkOS\Resource\ConnectionType $type The only supported ConnectionType at the time of this writing is MagicLink
+     *
+     * @return  \WorkOS\Resource\PasswordlessSession
+     */
+    public function createSession($email, $redirectUri, $state, $type)
+    {
+        $createSessionPath = "passwordless/sessions";
+    
+        $params = [
+            "email" => $email,
+            "type" => $type
+        ];
+
+        if ($redirectUri) {
+            $params["redirect_uri"] = $redirectUri;
+        }
+
+        if (null !== $state && !empty($state)) {
+            $params["state"] = \json_encode($state);
```

Comment:
Do we need to encode `$state`? We pass this through directly in our other SDKs (for [example](https://github.com/workos-inc/workos-go/blob/master/pkg/passwordless/client.go#L94)).

Reply:
Hmmm good point. I vaguely remember doing something similar to what I have here in the Python SDK but I see it's been changed over to accept a string too.

I think in hindsight, probably better to leave the encoding to the developer so I'll make the change here. The SSO class here still takes state as an associate array which I'll update in my next PR with all the small clientId/environment changes.